exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/BaseRootExec.java

```diff
@@ -123,6 +123,21 @@ public void receivingFragmentFinished(final FragmentHandle handle) {
     logger.warn("Currently not handling FinishedFragment message");
   }
 
+  @Override
+  public void dumpOperators() {
+    if (!operators.isEmpty()) {
+      logger.info("Operator dump started.");
+      for (CloseableRecordBatch batch : operators) {
+        batch.dump();
```

Comment:
You are right, I suppose a failed and its successful parent operators should do. Thanks.